src/Options/Generic.hs

```diff
@@ -326,40 +373,52 @@ instance ParseFields Data.Text.Lazy.Text
 instance ParseFields FilePath
 
 instance ParseFields Bool where
-    parseFields m =
+    parseFieldsH h m =
         case m of
             Nothing   -> do
                 let fs =  Options.metavar "BOOL"
+                       <> h
                 Options.argument auto fs
             Just name -> do
-                Options.switch (Options.long (Data.Text.unpack name))
+                Options.switch (Options.long (Data.Text.unpack name) <> h)
 
 instance ParseFields () where
-    parseFields _ = pure ()
+    parseFieldsH _ _ = pure ()
 
 instance ParseFields Any where
-    parseFields = fmap (fmap mconcat . many . fmap Any) parseField
+    parseFieldsH h m = (fmap mconcat . many . fmap Any) (parseFieldH h m)
 
 instance ParseFields All where
-    parseFields = fmap (fmap mconcat . many . fmap All) parseField
+    parseFieldsH h m = (fmap mconcat . many . fmap All) (parseFieldH h m)
 
 instance ParseField a => ParseFields (Maybe a) where
-    parseFields = fmap optional parseField
+    parseFieldsH h m = optional (parseFieldH h m)
 
 instance ParseField a => ParseFields (First a) where
-    parseFields = fmap (fmap mconcat . many . fmap (First . Just)) parseField
+    parseFieldsH h m = (fmap mconcat . many . fmap (First . Just)) (parseFieldH h m)
 
 instance ParseField a => ParseFields (Last a) where
-    parseFields = fmap (fmap mconcat . many . fmap (Last . Just)) parseField
+    parseFieldsH h m = (fmap mconcat . many . fmap (Last . Just)) (parseFieldH h m)
 
 instance (Num a, ParseField a) => ParseFields (Sum a) where
-    parseFields = fmap (fmap mconcat . many . fmap Sum) parseField
+    parseFieldsH h m = (fmap mconcat . many . fmap Sum) (parseFieldH h m)
 
 instance (Num a, ParseField a) => ParseFields (Product a) where
-    parseFields = fmap (fmap mconcat . many . fmap Product) parseField
+    parseFieldsH h m = (fmap mconcat . many . fmap Product) (parseFieldH h m)
 
 instance ParseField a => ParseFields [a] where
-    parseFields = parseListOfField
+    parseFieldsH = parseListOfFieldH
+
+newtype (<?>) (field :: *) (help :: Symbol) = Helped { getField :: field } deriving (Generic)
```

Comment:
Could you rename the `Helped` constructor to `Helpful`?  I believe "helpful" is the more accurate adjective for word "help"

Also, the accessor could be named `unHelpful` as a joke, but you don't have to go that far :)